angular/src/app/views/settings/settings.component.ts

```diff
@@ -99,88 +91,86 @@ export class SettingsComponent {
     );
   }
 
+  private loadOrderStatusValues(): void {}
+
   /**
    * Load the settings in an Object into the in-memory settings
    * @param data the settings as an Object
    */
-  private loadSettingsFromObject(data): void {
-    this.stockBuffer = data.stockBuffer;
-    this.defaultOrderStatus = data.defaultOrderStatus;
-    this.defaultShippingProvider = data.defaultShippingProvider;
+  private loadSettingsFromObject(data: SettingsInterface): void {
+    this.stockBuffer =
+      data.stockBuffer && data.stockBuffer > 0 ? data.stockBuffer : 0;
     this.defaultItemMappingMethod = data.defaultItemMappingMethod;
     this.importOrdersSince = data.importOrdersSince;
     this.isAllInventorySyncEnabled = data.isAllInventorySyncEnabled;
-  }
 
-  /**
-   * normalize the settings in memory,
-   * returning any errors as string
-   * @returns string
-   */
-  private normalizeSettings(): string {
-    if (this.importOrdersSince) {
-      this.importOrdersSince = new Date(this.importOrdersSince)
-        .toISOString()
-        .slice(0, 10);
-    }
+    this.chooseOrderStatusAfterRefreshingList(data.defaultOrderStatus);
+  }
 
-    return null;
+  private chooseOrderStatusAfterRefreshingList(statusId): void {
+    this.orderStatusService.fetch().subscribe(
+      (data) => {
+        // store this so that the UI can show in drop-down
+        this.orderStatuses = data;
+        if (statusId) {
+          if (this.orderStatuses.length > 0) {
+            for (let option of this.orderStatuses) {
+              if (option.statusId == statusId) {
+                this.defaultOrderStatus = statusId;
+                return;
+              }
+            }
+          }
+        }
+        // status ID not set or not a valid option - use default
+        this.defaultOrderStatus = SettingsComponent.DEFAULT_ORDER_STATUS_ID;
+      },
+      (err) => {
+        this.showErrorVerbose(this.translation.translate("error_fetch"));
+        return;
+      }
+    );
   }
 
   /**
-   * validate the settings in memory,
-   * returning any errors as string
-   * @returns string
+   * normalize the settings in memory so they're ready for the back-end
    */
-  private validateSettings(): string {
-    let issueStringBuffer = "";
-
-    if (this.stockBuffer < 0 || isNaN(this.stockBuffer)) {
-      issueStringBuffer +=
-        this.translation.translate("buffer") +
-        ": " +
-        this.translation.translate(
-          SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
-        ) +
-        SettingsComponent.MESSAGE_DELIM;
+  private normalizeSettings(): void {
+    if (this.importOrdersSince) {
+      try {
+        this.importOrdersSince = new Date(this.importOrdersSince)
```

Comment:
Does timezone matter here? 
`toISOString` default convert to UTC.
Would `importOrdersSince` be date only with local time zone(e.g. GMT+2)? 
If so, we could be possible off by 1 day

Reply:
FYI This is old code that just appears to be new due to being moved down later in the file, not actually something I'm trying to change in this PR.

That said, you are correct - there is a timezone issue that should be fixed.

Reply:
I'm going to fix this with the `moment` library:
![image](https://user-images.githubusercontent.com/44010975/90703374-994bbb80-e25b-11ea-9faf-4b1c6da4af77.png)

Reply:
huh, that made it worse, because the `terra-date-picker` assumes that `2020-08-18` without any timezone is in GMT.

Reply:
the `terra-date-picker` element is awful. I'm switching to a normal `input` with type `date` - plays nicer with browsers and only uses "YYYY-MM-DD" format.

Reply:
Yeah this is much better - now the browser knows it's a date so you get your client's preferred calendar interface: 

![image](https://user-images.githubusercontent.com/44010975/90708746-87244a00-e268-11ea-8a57-079c6465a1c2.png)

Reply:
Neat!